core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -43,6 +42,18 @@ private[spark] class ApplicationInfo(
   @transient var endTime: Long = _
   @transient var appSource: ApplicationSource = _
 
+  // A cap on the number of executors this application can have at any given time.
+  // By default, this is infinite. Only after the first allocation request is issued
+  // by the application will this be set to a finite value.
+  @transient var executorLimit: Int = _
```

Comment:
You could initialize it here to `Int.MaxValue`. `_` is 0, and the comment is misleading, since there is no such thing as infinite values for integers.